src/Services/SubmitService/SubmitService.cs

```diff
@@ -79,21 +88,59 @@ public async Task<string> ProcessSubmission(MappingEntity mappingEntity, string
                 reference = answers.CaseReference;
             }
 
-            return _submissionServiceConfiguration.FakeSubmission 
-                ? ProcessFakeSubmission(mappingEntity, form, sessionGuid, reference) 
+            if (baseForm.Pages is not null && mappingEntity.FormAnswers.Pages is not null)
+            {
+                var journeyPages = baseForm.GetReducedPages(mappingEntity.FormAnswers);
+
+                var isAutoConfirm = journeyPages.Any(_ => _.Elements.Any(_ => _.Type.Equals(EElementType.Booking) && _.Properties.AutoConfirm.Equals(true)));
+                if (isAutoConfirm)
+                    return await ConfirmBooking(mappingEntity, form, sessionGuid, baseForm, reference);
+            }
+
+            return _submissionServiceConfiguration.FakeSubmission
+                ? ProcessFakeSubmission(mappingEntity, form, sessionGuid, reference)
                 : await ProcessGenuineSubmission(mappingEntity, form, sessionGuid, baseForm, reference);
         }
 
+        private async Task<string> ConfirmBooking(MappingEntity mappingEntity, string form, string sessionGuid, FormSchema baseForm, string reference)
+        {
+            var bookingProperties = baseForm.Pages
+                               .Where(page => page.Elements is not null)
+                               .SelectMany(page => page.Elements)
+                               .First(element => element.Type.Equals(EElementType.Booking)).Properties;
+
+            var bookingId = Convert.ToString(mappingEntity.FormAnswers.AllAnswers
+                .ToDictionary(x => x.QuestionId, x => x.Response)
+                .Where(_ => _.Key.Contains(BookingConstants.RESERVED_BOOKING_ID))
+                .FirstOrDefault().Value);
```

Comment:
this might be an issue when multiple booking are used, are we able to support multiple booking elements straight away?

Reply:
The scope of this work is to handle one one auto completed submission. We can look into auto completing multiple if/when a form comes up that needs this functionality

Reply:
the linq will need updating as u might selected a different booking answers if the form has multiple, based on it doign Contains

Reply:
True, I don't know how many forms will have multiple bookings on the same form but yes, if we can amend the LINQ to handle that we cover our own backs 👍